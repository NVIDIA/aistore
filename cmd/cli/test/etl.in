// RUN k8s
curl https://raw.githubusercontent.com/NVIDIA/ais-etl/main/transformers/hello_world/pod.yaml -o hello_world_init_spec.yaml -s // IGNORE
ais etl init spec --from-file="hello_world_init_spec.yaml" --comm-type="hpush://" --name=$ETL_NAME-1 // IGNORE
ais etl show | grep $ETL_NAME-1 | awk '{print $2}'
ais etl init spec --from-file="hello_world_init_spec.yaml" --comm-type="hpush://" --name=$ETL_NAME-2 --arg-type="fqn" // IGNORE
ais etl show | grep $ETL_NAME-2 | awk '{print $2}'
rm -f hello_world_init_spec.yaml // IGNORE

// Test URL support for --from-file flag (new functionality)
ais etl init spec --from-file="https://raw.githubusercontent.com/NVIDIA/ais-etl/main/transformers/hash_with_args/pod.yaml" --comm-type="hpush://" --name=$ETL_NAME-url // IGNORE
ais etl show | grep $ETL_NAME-url | awk '{print $2}'

curl https://raw.githubusercontent.com/NVIDIA/ais-etl/main/transformers/hash_with_args/pod.yaml -o hash_with_args_init_spec.yaml -s // IGNORE
ais etl init spec --from-file="hash_with_args_init_spec.yaml" --comm-type="hpush://" --name=$ETL_NAME-3 // IGNORE
ais etl show | grep $ETL_NAME-3 | awk '{print $2}'
rm -f hash_with_args_init_spec.yaml // IGNORE

echo "0123456789" > $OBJECT_1.txt
ais bucket create ais://$BUCKET_1 | awk '{print $1 " " $2}'
ais object put $OBJECT_1.txt ais://$BUCKET_1/object_1.txt
rm -f $OBJECT_1.txt // IGNORE

ais etl show details $ETL_NAME-1 | grep "COMMUNICATION TYPE"
ais etl show details $ETL_NAME-1 | grep "ARGUMENT"
ais etl object $ETL_NAME-1 ais://$BUCKET_1/object_1.txt -
ais etl stop $ETL_NAME-1
ais etl show | grep $ETL_NAME-1 | awk '{print $2}'
ais etl start $ETL_NAME-1
ais etl object $ETL_NAME-1 ais://$BUCKET_1/object_1.txt -
ais etl bucket $ETL_NAME-1 ais://$BUCKET_1 ais://$BUCKET_2 --list '$OBJECT_1.txt,$OBJECT_2.txt' // IGNORE
ais etl show errors $ETL_NAME-1 -H | awk {'print $1'}
ais etl stop $ETL_NAME-1
ais etl rm $ETL_NAME-1

ais etl show details $ETL_NAME-2 | grep "COMMUNICATION TYPE"
ais etl show details $ETL_NAME-2 | grep "ARGUMENT"
ais etl object $ETL_NAME-2 ais://$BUCKET_1/object_1.txt -
ais etl stop $ETL_NAME-2
ais etl rm $ETL_NAME-2

// Test URL-loaded ETL
ais etl show details $ETL_NAME-url | grep "COMMUNICATION TYPE"
ais etl show details $ETL_NAME-url | grep "ARGUMENT"
ais etl object $ETL_NAME-url ais://$BUCKET_1/object_1.txt - --args=url-test
ais etl stop $ETL_NAME-url
ais etl rm $ETL_NAME-url

ais etl show details $ETL_NAME-3 | grep "COMMUNICATION TYPE"
ais etl show details $ETL_NAME-3 | grep "ARGUMENT"
ais etl object $ETL_NAME-3 ais://$BUCKET_1/object_1.txt - --args=123
ais etl object $ETL_NAME-3 ais://$BUCKET_1/object_1.txt - --args=456
ais etl stop $ETL_NAME-3
ais etl rm $ETL_NAME-3