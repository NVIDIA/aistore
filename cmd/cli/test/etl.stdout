Running
Running
Running
^*.ais://$BUCKET_1*.created$
^PUT.*=> ais://$BUCKET_1/object_1.txt$

COMMUNICATION TYPE:  hpush://
ARGUMENT TYPE:
Hello World!
^ETL\[etl.*\]stopped$
Stopped
^ETL\[etl.*\]started successfully$
Hello World!
^ETL\[etl.*\]stopped$
^ETL\[etl.*\] successfully deleted$
COMMUNICATION TYPE:  hpush://
ARGUMENT TYPE: fqn
Hello World!
$OBJECT_2.txt
^ETL\[etl.*\]stopped$
^ETL\[etl.*\] successfully deleted$
COMMUNICATION TYPE:  hpush://
ARGUMENT TYPE:
5328b3c272b5aab0
aecb71fa663c6460
^ETL\[etl.*\]stopped$
^ETL\[etl.*\] successfully deleted$
